### PR TITLE
Update eslint v9

### PR DESCRIPTION
**Pull request by liuxingbaoyu**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -528,7 +528,7 @@ jobs:
       - name: Downgrade ESLint to 7.5.0
         run: yarn up eslint@7.5.0
       - name: Run babel/eslint tests
-        run: node ./node_modules/.bin/jest eslint
+        run: node ./node_modules/.bin/jest eslint/babel-eslint-parser/ eslint/babel-eslint-plugin/ babel-eslint-tests/
```
Comment (nicolo-ribaudo) [inline .github/workflows/ci.yml:531]: Why do we need this change?

**Review by nicolo-ribaudo — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (liuxingbaoyu) [reply .github/workflows/ci.yml:531]: https://github.com/babel/babel/actions/runs/8980473754/job/24664241688
Our internal plugins no longer support the old eslint.

Comment (liuxingbaoyu) [reply .github/workflows/ci.yml]: `eslint-plugin-development` appears to be public, despite only having a few hundred weekly downloads, should we keep compatibility?

**Review by JLHwung — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nicolo-ribaudo) [reply .github/workflows/ci.yml]: Whops, I missed this. Yes, we should keep compatibility.

**Pull request merged**